File: 1.0/src/cz/mg/vulkan/PFNvkMapMemory.java
package cz.mg.vulkan;

public class PFNvkMapMemory extends VkFunctionPointer {
    public PFNvkMapMemory() {
    }

    protected PFNvkMapMemory(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkMapMemory(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkMapMemory(long value) {
        setValue(value);
    }

    public PFNvkMapMemory(VkInstance instance) {
        super(instance, new VkString("vkMapMemory"));
    }

    public void call(VkDevice device, VkDeviceMemory memory, VkDeviceSize offset, VkDeviceSize size, VkMemoryMapFlags flags, VkObject ppData, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), memory != null ? memory.getVkAddress() : VkPointer.getNullAddressNative(), offset != null ? offset.getVkAddress() : VkPointer.getNullAddressNative(), size != null ? size.getVkAddress() : VkPointer.getNullAddressNative(), flags != null ? flags.getVkAddress() : VkPointer.getNullAddressNative(), ppData != null ? ppData.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long memory, long offset, long size, long flags, long ppData, long rval);


    public int call(VkDevice device, VkDeviceMemory memory, long offset, long size, int flags, VkObject ppData){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), memory != null ? memory.getVkAddress() : VkPointer.getNullAddressNative(), offset, size, flags, ppData != null ? ppData.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long memory, long offset, long size, int flags, long ppData);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkMapMemory.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkMapMemory_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong memory, jlong offset, jlong size, jlong flags, jlong ppData, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkMapMemory f = (PFN_vkMapMemory)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDeviceMemory*)jniLongToPointer(memory)),
        *((VkDeviceSize*)jniLongToPointer(offset)),
        *((VkDeviceSize*)jniLongToPointer(size)),
        *((VkMemoryMapFlags*)jniLongToPointer(flags)),
        ((void**)jniLongToPointer(ppData))
    );
}


jint Java_cz_mg_vulkan_PFNvkMapMemory_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong memory, jlong offset, jlong size, jint flags, jlong ppData){
    (void)env;
    (void)jc;
    PFN_vkMapMemory f = (PFN_vkMapMemory)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDeviceMemory*)jniLongToPointer(memory)),
        (VkDeviceSize)offset,
        (VkDeviceSize)size,
        (VkMemoryMapFlags)flags,
        ((void**)jniLongToPointer(ppData))
    );
}

